client/src/actions/data/dataActions.js

```diff
@@ -74,3 +75,18 @@ export const fetchCategory = (categoryId = null) => async dispatch => {
     apiActionExceptionHandler(e, dispatch);
   }
 };
+
+export const addProduct = (productId = null) => async dispatch => {
+  if (!productId) throw Error('missing productId!');
+  dispatch(clearAllErrors());
+  dispatch({ type: DATA_LOADING, payload: true });
+
+  try {
+      const res = await axios.post(`/api/add/`);
+      dispatch({ type: DATA_ADD_PRODUCT, payload: res.data });
+      
+      
+  } catch (e) {
+      
```

Comment:
look at the other handlers, we still need to cater for the exception, so add:

```suggestion
dispatch(dataLoadingStop());
apiActionExceptionHandler(e, dispatch);
```

in here a well